client/src/modules/roles/modal/rolesPermissions.js

```diff
@@ -0,0 +1,69 @@
+angular.module('bhima.controllers')
+  .controller('RolesPermissionsController', RolesPermissionsController);
+
+RolesPermissionsController.$inject = [
+  'data', '$state', '$uibModalStack', 'AppCache', 'RolesService', 'NotifyService'
+];
+
+function RolesPermissionsController(data, $state, $uibModalStack, AppCache, RolesService, Notify) {
+  var vm = this;
+  vm.close = close;
+  vm.role = angular.copy(data);
+  vm.tree = [];
+  vm.selectAll = selectAll;
+  vm.allPage = 0;
+
+  // affeted pages(permission) to this role
+  vm.getAffected = getAffected;
+
+  // vm.role.uuid
+  RolesService.unit(vm.role.uuid)
+    .then(
+      function (res) {
+        vm.tree = res.data;
+      });
+
+  // close modal
+  function close() {
+    $uibModalStack.dismissAll();
+  }
+
+  function selectAll() {
+
+    for (var i = 0; i < vm.tree.length; i++) {
+      var _module = vm.tree[i];
+      _module.affected = vm.allPage;
+      
+      for (var j = 0; j < _module.pages.length; j++) {
```

Comment:
I know but it seem to be an array function, right?

Reply:
Ah, array functions are fine.  They are in ES5.  It's "fat arrow" functions that you don't want to use until #2472 is fulfilled.  So `Array.prototype.forEach()` is fine and `Array.prototype.map()` and the rest are okay.

The thing that _doesn't_ work is `var x = (y) => y + 1;`.  That is an "arrow function" since it is a function that doesn't use the word `function`.  Those are ES2015+.